### PR TITLE
TS-4979: Fix clang-analyzer complaint about potential null pointer.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**

### CODE REVIEW

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/Github-Linux/930/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1039/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/931/ for details.

File: proxy/http/HttpTransact.cc [hunk 1]
```diff
@@ -1075,6 +1073,8 @@ HttpTransact::ModifyRequest(State *s)
   if (!request.is_target_in_url()) {
     s->hdr_info.client_req_is_server_style = true;
   }
+  // Make Coverity happy. hostname is non-null iff request.is_target_in_url().
```
Comment (zwoop) [inline proxy/http/HttpTransact.cc]: Since we're bike shedding, this probably should have said "clang-analyzer". ;-)

Comment (SolidWallOfCode) [inline proxy/http/HttpTransact.cc]: It's just who we are.

**Pull request merged**